src/solver/conv_mlir_igemm_fwd.cpp

```diff
@@ -39,8 +39,9 @@ namespace solver {
 
 const PerformanceConvMlirIgemm& PerformanceConvMlirIgemm::MlirHeuristicInitRequest()
 {
+    // HeuristicInit is "spare" in nature
     static const PerformanceConvMlirIgemm p =
-        PerformanceConvMlirIgemm(-2, -2, -2, -2, -2, -2, false);
+        PerformanceConvMlirIgemm(-2, -2, -2, -2, -2, -2, false, true);
```

Comment:
Please use the same approach in `ConvMlirIgemmFwd::GetPerformanceConfig()`.

Reply:
Adopted all other review comments but this one: The temporary `heur` object will not have access to its private method. Also, this will make below impossible if private:

https://github.com/ROCmSoftwarePlatform/MIOpen/blob/223b4b814676809274e105df70749fd189059a55/src/include/miopen/solver/mlir_common.hpp#L52-L53

Therefore I kept it as-is.

Reply:
> The temporary heur object will not have access to its private method.

Then let's make it public. The https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1435#discussion_r831382022 is very important and must be resolved.

> Also, this will make below impossible if private...

Well, then we have to rework other MLIR solvers. This can be done in the future. Right here you can use specialization of `ConstructBuildOptions` for the solvers affected by this PR). The approach is the same: create temporary object, initialize it with "heuristic init request" and then compare.

Reply:
@junliume @jerryyin [Notice on process] I do not recommend marking comments as resolved until there is consensus. Why: contributors (like me) can not "unresolve" comments.

Reply:
> Please initialize only necessary members to -2; do not touch others and return void. Let's also make this method private.

Okay I think I got what you mean now. I'll not use the constructor to do the initialization. Instead I'll just set the fields and not return.

Reply:
To be honest, by doing below:

```cpp
    PerformanceConvMlirIgemm heur;
    heur.MlirHeuristicInitRequest();
    if(*this == heur)
        return true;
```

You are implicitly leaving the `use_spare_set` uninitialized because the default constructor does not set that field. I don't like leaving field uninitialized because ultimately `use_spare_set` still have a value out there. I'd actually prefer to original method of setting the `use_spare_set` explicitly because the heuristic entry is a "spare" entry.

Reply:
@atamazov On issue resolution, do you have the permission to resolve an issue? If not, the other way around is that I can unresolve any issue that you want further discussion as long as you leave new comments to it. Either works for me because I also don't want to accidentally suppressed important comments.

Reply:
Done in [2c17fdd](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1435/commits/2c17fdd8d77360cbaa72c6a9b276bd2f5c027d8a)